### ISSUE TITLE
Add topologySpreadConstraints to helm chart

### ISSUE THREAD

**Issue by etiennep** [grafana/agent#5939]
### Request

I would like to have a topologySpreadConstraints configurable setting added to the Grafana Agent helm chart. Settings must be configurable through the values.yaml.

### Use case

When deploying clustered Grafana Agents with a Deployment or StatefulSet, it would be useful to have topologySpreadConstraints to the pod specs so that pods get distributed evenly across multiple availability zones. 

This is especially important when using karpenter to provision nodes because pods might all get scheduled on a single instance, which isn't great for system resiliency if that one node goes away.

**Comment by tpaschalis** [comment]
Sounds like a useful feature to me! Would you be interested in contributing this?

It'd include adding the new option in the templates and values file of our [helm directory](https://github.com/grafana/agent/tree/main/operations/helm/charts/grafana-agent), as well as adding a new test values.yaml file in the ci folder

**Comment by etiennep** [comment]
Yes I can work on this. It's easy enough and something we could use today. Feel free to assign.
